packaging/installer/netdata-uninstaller.sh

```diff
@@ -24,422 +24,450 @@ ENVIRONMENT_FILE="/etc/netdata/.environment"
 
-	for p in \
-		$(cat /var/run/netdata.pid 2>/dev/null) \
-		$(cat /var/run/netdata/netdata.pid 2>/dev/null) \
-		$(safe_pidof netdata 2>/dev/null); do
-		ns="$(readlink "/proc/${p}/ns/pid" 2>/dev/null)"
+  for p in \
+    $(cat /var/run/netdata.pid 2> /dev/null) \
+    $(cat /var/run/netdata/netdata.pid 2> /dev/null) \
+    $(safe_pidof netdata 2> /dev/null); do
+    ns="$(readlink "/proc/${p}/ns/pid" 2> /dev/null)"
 
-		if [ -z "${myns}" ] || [ -z "${ns}" ] || [ "${myns}" = "${ns}" ]; then
-			pidisnetdata "${p}" && echo "${p}"
-		fi
-	done
+    if [ -z "${myns}" ] || [ -z "${ns}" ] || [ "${myns}" = "${ns}" ]; then
+      pidisnetdata "${p}" && echo "${p}"
+    fi
+  done
 }
 
 stop_all_netdata() {
-	local p
-
-	if [ "${UID}" -eq 0 ] ; then
-		uname="$(uname 2>/dev/null)"
-
-		# Any of these may fail, but we need to not bail if they do.
-		if issystemd; then
-			if systemctl stop netdata ; then
-				sleep 5
-			fi
-		elif [ "${uname}" = "Darwin" ]; then
-			if launchctl stop netdata ; then
-				sleep 5
-			fi
-		elif [ "${uname}" = "FreeBSD" ]; then
-			if /etc/rc.d/netdata stop ; then
-				sleep 5
-			fi
-		else
-			if service netdata stop ; then
-				sleep 5
-			fi
-		fi
-	fi
-
-	if [ -n "$(netdata_pids)" ] &&  [ -n "$(builtin type -P netdatacli)" ] ; then
-		netdatacli shutdown-agent
-		sleep 20
-	fi
-
-	for p in $(netdata_pids); do
-		# shellcheck disable=SC2086
-		stop_netdata_on_pid ${p}
-	done
+  local p
+
+  if [ "${UID}" -eq 0 ]; then
+    uname="$(uname 2> /dev/null)"
+
+    # Any of these may fail, but we need to not bail if they do.
+    if issystemd; then
+      if systemctl stop netdata; then
+        sleep 5
+      fi
+    elif [ "${uname}" = "Darwin" ]; then
+      if launchctl stop netdata; then
+        sleep 5
+      fi
+    elif [ "${uname}" = "FreeBSD" ]; then
+      if /etc/rc.d/netdata stop; then
+        sleep 5
+      fi
+    else
+      if service netdata stop; then
+        sleep 5
+      fi
+    fi
+  fi
+
+  if [ -n "$(netdata_pids)" ] && [ -n "$(builtin type -P netdatacli)" ]; then
+    netdatacli shutdown-agent
+    sleep 20
+  fi
+
+  for p in $(netdata_pids); do
+    # shellcheck disable=SC2086
+    stop_netdata_on_pid ${p}
+  done
 }
 
 trap quit_msg EXIT
 
+uninstall_ebpf() {
+  # Removing NetData's libexec directory is taken care of which is where we
+  # installed the rest of the shared libraires for the eBPF Collector.
+
+  echo >&2 " Finding lib directory ..."
+  libdir=
+  libdir="$(ldconfig -v 2> /dev/null | grep ':$' | sed -e 's/://' | sort -r | grep 'usr' | head -n 1)"
+  if [ -z "${libdir}" ]; then
+    libdir="$(ldconfig -v 2> /dev/null | grep ':$' | sed -e 's/://' | sort -r | head -n 1)"
+  fi
+
+  if [ -z "${libdir}" ]; then
+    echo >&2 "Unable to find the system lib directory we installed eBPF ebpf_kernel.so to ..."
+    user_input "Press ENTER to search your system (which may take some time) ..."
+    find / -type f -name 'bpf_kernel.so' -o -name 'bpf_kernel.so.o'
```

Comment:
I understand that find can search for partial names, but should not we use the complete name( `libbpf_kernel.so` instead `bpf_kernel.so.`?
Another thing, please, change `bpf_kernel.so.o` to `bpf_kernel.so.0`, the second name is wrong.

Reply:
Thanks Fixed :)